xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainAdapters.java

```diff
@@ -0,0 +1,256 @@
+            return (CurrencyPair) instrument;
+        }
+        throw new IllegalArgumentException(String.format("Unsupported instrument '%s'", instrument));
+    }
+
+    public static Ticker toTicker(BlockchainTicker blockchainTicker) {
+        return new Ticker.Builder()
+                .instrument(blockchainTicker.getSymbol())
+                .last(blockchainTicker.getLastTradePrice())
+                .volume(blockchainTicker.getVolume())
+                .build();
+    }
+
+    public static OrderBook toOrderBook(BlockchainOrderBook blockchainOrderBook) {
+        List<LimitOrder> asks = blockchainOrderBook.getAsks().stream()
+                .map(limitOrder -> toLimitOrder(limitOrder, Order.OrderType.ASK, blockchainOrderBook.getSymbol()))
+                .collect(Collectors.toList());
+        List<LimitOrder> bids = blockchainOrderBook.getAsks().stream()
+                .map(limitOrder -> toLimitOrder(limitOrder, Order.OrderType.BID, blockchainOrderBook.getSymbol()))
+                .collect(Collectors.toList());
+
+        return new OrderBook(null, asks, bids);
+    }
+
+    public static LimitOrder toLimitOrder(BlockchainMarketDataOrder blockchainMarketDataOrder, Order.OrderType orderType , CurrencyPair currencyPair) {
+        return new LimitOrder.Builder(orderType, currencyPair)
+                .instrument(currencyPair)
+                .limitPrice(blockchainMarketDataOrder.getPrice())
+                .originalAmount(blockchainMarketDataOrder.getQuantity())
+                .build();
+    }
+
+    public static Trades toTrades(List<BlockchainTrade> blockchainTrades, CurrencyPair currencyPair) {
+        List<Trade> trades = blockchainTrades.stream()
+                .map(blockchainTrade -> {
+                    Order.OrderType orderType = blockchainTrade.isBuyer() ? Order.OrderType.BID : Order.OrderType.ASK;
+                    return new Trade.Builder()
+                            .instrument(currencyPair)
+                            .type(orderType)
+                            .originalAmount(blockchainTrade.getQuantity())
+                            .price(blockchainTrade.getPrice())
+                            .timestamp(blockchainTrade.getTimestamp())
+                            .id(blockchainTrade.getId())
+                            .build();
+                }).collect(Collectors.toList());
+        return new Trades(trades, Trades.TradeSortType.SortByTimestamp);
+    }
+
+    public static Integer currencyPairToId(CurrencyPair currencyPair) {
+        if (CurrencyPair.ETH_BTC.equals(currencyPair)) {
+            return 3;
+        }
+        throw new IllegalArgumentException(String.format("Unsupported currency pair '%s'", currencyPair));
+    }
+
+
+    public static AddressWithTag toAddressWithTag(BlockchainAccount blockchainAccount){
+        return new AddressWithTag(blockchainAccount.getAddress(), null);
+    }
+
+    public static Order.OrderStatus toOrderStatus(String status) {
```

Comment:
I think you should use valueOf:

`        try {
           return Order.OrderStatus.valueOf(status.toUpperCase())
        } catch (IllegalArgumentException e) {
            return Order.OrderStatus.UNKNOWN;
        }`

Reply:
https://stackoverflow.com/questions/604424/how-to-get-an-enum-value-from-a-string-value-in-java

Reply:
This was implented for the poc, for the moment it will be deleted for this PR.